src/mesh/python/x3d_generator.py

```diff
@@ -51,6 +59,65 @@
 elif args.mesh_type in ['vor_2d_mesh']:
     mesh = mesh_type_dict[args.mesh_type](bnd_per_dim, args.num_cells)
 
+# ------------------------------------------------------------------------------------------------ #
+# -- construct matid and region dictionary if region data arguments are available
+
+# -- set matid data (using region ids and bounds if available)
+matids = np.zeros(mesh.num_cells, dtype=int)
+reg_dict = {}
+reg_ids = []
+if (len(args.reg_ids) > 0):
+    # -- prepend reg_ids and reg_bnd_per_dim with universal region
+    reg_ids = args.reg_ids
+    reg_bnd_per_dim = args.reg_bnd_per_dim
+    reg_ids = [0] + reg_ids
+    reg_bnd_per_dim = args.bnd_per_dim + reg_bnd_per_dim
+    # -- initialize dictionary of cell lists
+    reg_dict = {reg_id: [] for reg_id in reg_ids}
+    nmats = len(reg_ids)
+    for cell in range(mesh.num_cells):
+        # -- initialize container for cell average coordinate
+        coord_av = np.array([0.0, 0.0, 0.0])
```

Comment:
This cell-average coordinate computation should probably be a method for each mesh (for Voronoi it would just be the original sampled points used to generate the mesh).  However, this would be a fairly minor change (and just a refactor, not a significant feature change).